pkg/storage/chunk/client/object_client.go

```diff
@@ -112,7 +116,8 @@ func (o *client) PutWal(ctx context.Context, segment *wal.SegmentWriter) error {
 	if err != nil {
 		return err
 	}
-	return o.store.PutObject(ctx, "wal-segment-"+time.Now().UTC().Format(time.RFC3339Nano), bytes.NewReader(buffer.Bytes()))
+	newUlid := ulid.MustNew(ulid.Timestamp(time.Now()), o.entropy)
```

Comment:
I wonder if like Pyroscope using rand.Reader isn't enough ? https://github.com/grafana/pyroscope/blob/3fb6d31287dceeb83054f47a23a24d57fdb17ab3/pkg/phlaredb/block/metadata.go#L242

Reply:
Nice, they use `crypto/rand` and it avoids needing to carry a reader in the struct so I'll just use that.